src/NLog/SetupExtensionsBuilderExtensions.cs

```diff
@@ -142,9 +144,18 @@ public static ISetupExtensionsBuilder RegisterLayoutRenderer(this ISetupExtensio
         /// <param name="setupBuilder">Fluent interface parameter.</param>
         /// <param name="name">Name of the layout renderer - without ${}.</param>
         /// <param name="layoutMethod">Callback that returns the value for the layout renderer.</param>
-        public static ISetupExtensionsBuilder RegisterLayoutRenderer(this ISetupExtensionsBuilder setupBuilder, string name, Func<LogEventInfo, LoggingConfiguration, object> layoutMethod)
+        /// <param name="options">Options of the layout renderer.</param>
+        public static ISetupExtensionsBuilder RegisterLayoutRenderer(this ISetupExtensionsBuilder setupBuilder, string name, Func<LogEventInfo, LoggingConfiguration, object> layoutMethod, LayoutRenderOptions options = LayoutRenderOptions.None)
         {
-            var layoutRenderer = new FuncLayoutRenderer(name, layoutMethod);
+            FuncLayoutRenderer layoutRenderer;
+
+            if ((options & LayoutRenderOptions.ThreadAgnostic) == LayoutRenderOptions.ThreadAgnostic)
```

Comment:
Possible something could be done to reuse the logic from Layout.CreateFuncLayoutRenderer. Not good with duplicate code.